test/utils/pod/pod_utils.go

```diff
@@ -124,10 +126,33 @@ func WaitForPodsWithMatchingLabelToRun(testTimeout int, tickerDuration time.Dura
 		case <-ticker.C:
 			tickerCount++
 			numberOfRunningPods = CountAllPodsWithMatchingLabel(cs, ns, map[string]string{"group": "azuredisk-scale-tester"}, string(v1.PodRunning))
-			e2elog.Logf("%d min: %d pods are ready", tickerCount, numberOfRunningPods)
+			e2elog.Logf("%.1f min: %d pods are running", float64(tickerCount)*tickerDuration.Seconds()/60, numberOfRunningPods)
 			if numberOfRunningPods >= desiredNumberOfPods {
 				return
 			}
 		}
 	}
 }
+
+// WaitForAllPodsWithMatchingLabelToDelete waits for all pods to be deleted until timeout occurs, whichever comes first,
+// and returns the number of running pods.
+func WaitForAllPodsWithMatchingLabelToDelete(testTimeout int, tickerDuration time.Duration, cs clientset.Interface, ns *v1.Namespace) (numberOfRunningPods int) {
+	framework.ExpectEqual(testTimeout > int(tickerDuration.Minutes()), true)
+	ticker := time.NewTicker(tickerDuration)
+	tickerCount := 0
+	timeout := time.After(time.Duration(testTimeout) * time.Minute)
+
+	for {
+		select {
+		case <-timeout:
+			return
+		case <-ticker.C:
+			tickerCount++
+			numberOfRunningPods = CountAllPodsWithMatchingLabel(cs, ns, map[string]string{"group": "azuredisk-scale-tester"}, string(v1.PodRunning))
+			e2elog.Logf("%.1f min: %d pods are running", float64(tickerCount)*tickerDuration.Seconds()/60, numberOfRunningPods)
```

Comment:
np: can we get the duration in minutes here directly?

Reply:
Oh yes!